queries/go.scm

```diff
@@ -35,3 +35,146 @@
+;;       - What is the type of T{1: 2}? It could be array, map, or slice.
+;;       - What about T{a: 1}? It could be map or struct.
+;;       - What about T{1, 2}? It could be struct, array, or slice.
+;;     Cursorless only has syntactic information available to it.
+;;
+;;   * The user might not know the type either. With a named type, the type definition might be far away.
+;;     Or it might just be offscreen. Either way, the user needs to be able to make a decision about
+;;     what scope to use using only locally available, syntactic information.
+;;     Note that this also means that has-a predicates work better than has-no predicates.
+;;     The user can locally confirm that there is a keyed element.
+;;     She cannot confirm locally that there is no keyed element; it might just not be visible.
+;;
+;; Combining all these constraints suggests the following simple rules:
+;;
+;;   * If there is a keyed element present, then it is a map.
+;;   * If there is a non-keyed element present, then it is a list.
+;;   * If there are both or neither, then it is both a map and a list.
+;;
+;; Conveniently, this is also simple to implement.
+;;
+;; This guarantees that a user always knows how to refer to any composite literal.
+;; There are cases in which being overgenerous in matching is not ideal,
+;; but they are rarer, so let's optimize for the common case.
+;; Mixed keyed and non-keyed elements are also rare in practice.
+;; The main ambiguity is with {}, but there's little we can do about that.
+;;
+;; Go users also expect that the map and list scopes will include the type definition,
+;; as well as any & before the type. (Strictly speaking it is not part of the literal,
+;; but that's not how most humans think about it.)
+;;
+;; If you are considering changing the map and list scopes, take a look at the examples in
+;; data/playground/go/maps_and_lists.go, which cover a fairly wide variety of cases.
+
+;; maps
+
+;; &T{a: 1}
+(unary_expression
+  operator: "&"
+  (composite_literal
+    body: (literal_value
+      (keyed_element)
+    )
+  )
+) @map
+
+;; T{a: 1}
+(
+  (composite_literal
+    body: (literal_value
+      (keyed_element)
+    )
+  ) @map
+  (#not-parent-type? @map unary_expression)
+)
+
+;; {a: 1}
+(
+  (literal_value
+    (keyed_element)
+  ) @map
+  (#not-parent-type? @map composite_literal)
+)
+
+;; lists
+
+;; &T{1}
+(unary_expression
+  operator: "&"
+  (composite_literal
+    body: (literal_value
+      (literal_element)
+    )
+  )
+) @list
+
+;; T{1}
+(
+  (composite_literal
+    body: (literal_value
+      (literal_element)
+    )
+  ) @list
+  (#not-parent-type? @list unary_expression)
+)
+
+;; {1}
+(
+  (literal_value
+    (literal_element)
+  ) @list
+  (#not-parent-type? @list composite_literal)
+)
+
+;; empty composite literals
+
+;; &T{}
+(unary_expression
+  operator: "&"
+  (composite_literal
+    body: (literal_value
+      .
+      "{"
+      .
+      "}"
+      .
+    )
+  )
+) @list @map
```

Comment:
Nice catch, and definitely something to think about in the future as I go. In this particular case I think the problem is going to be extremely rare, because  I would wager that comments inside `{}`  Are all but nonexistent. Given that the fix is very verbose, I think I'd just like to let leave this one be.

Reply:
I would certainly be happy to add a comment to that effect if you'd like. ( Or fix it if you care deeply.)

Reply:
I guess one kind of hacky implementation would be `(#count-non-comment-children 2)`.

Reply:
I'm fine with a one-line comment indicating it's not handled

Reply:
It's three short lines. You found my kryptonite :)